### PR TITLE
JSON support added to WebUI

### PR DESCRIPTION
**Pull request by tysonjh**
Accessing either the Worker or Master WebUI with a query parameter "?format=json" will yield a response with content-type "application/json" and a properly formatted JSON object.

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -133,6 +133,8 @@ object SparkBuild extends Build {
       "colt" % "colt" % "1.2.0",
       "cc.spray" % "spray-can" % "1.0-M2.1",
       "cc.spray" % "spray-server" % "1.0-M2.1",
+      "cc.spray" %%  "spray-json" % "1.1.1",
+      "io.spray" %%  "spray-json" % "1.2.3",
```
Comment (mateiz) [inline project/SparkBuild.scala]: Why do you add both of these in here? If you want we can change Spray altogether to io.spray.

Comment (tysonjh) [inline project/SparkBuild.scala]: There is no need for both here, you are correct. 

Using io.spray would mean spray-server would be replaced with spray-routing. Currently they are on release 1.1-M7 which is a "double upgrade" for supporting new Scala (2.9.2/2.10) and Akka (2.0.4) versions so may be worth waiting for a stable release to make the change.

**Comment by mateiz** [comment]
Cool, thanks! Looks like this doesn't show executor info but it's still useful.

**Comment by tysonjh** [comment]
I'm adding support right now for the ExecutorRunner, JobDescription classes so we can add the list of running/finished executors.

**Pull request merged**